### PR TITLE
feat: add ArbAggregator decorator

### PR DESCRIPTION
**Pull request by TucksonDev**
This PR adds a decorator to use the ArbAggregator precompile

### CODE REVIEW

File: src/arbAggregatorPrepareTransactionRequest.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  Transport,
+} from 'viem';
+
+import { arbAggregator } from './contracts';
+import { upgradeExecutorEncodeFunctionData } from './upgradeExecutor';
+import { Prettify } from './types/utils';
+
+type ArbAggregatorEncodeFunctionDataParameters = Prettify<
+  Omit<EncodeFunctionDataParameters<typeof arbAggregator.abi, string>, 'abi'>
+>;
```
Comment (chrstph-dvx) [inline src/arbAggregatorPrepareTransactionRequest.ts:16]: using `string` here means we won't infer `args` based on the function name. You can have a look at `arbOwnerPrepareTransactionRequest`

Comment (TucksonDev) [reply src/arbAggregatorPrepareTransactionRequest.ts]: Thanks Chris!
It looks like I copied the previous code when creating the decorator, and it definitely looks nicer now. I've made the necessary changes (waiting for the tests to finish)

File: src/decorators/arbAggregatorActions.integration.test.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        account: l2RollupOwner.address,
+      });
+    await nitroTestnodeL2Client.sendRawTransaction({
+      serializedTransaction: await l2RollupOwner.signTransaction(setFeeCollectorTransactionRequest),
+    });
+
+    // Check the fee collector has changed
+    const batchPosterFeeCollector = await nitroTestnodeL2Client.arbAggregatorReadContract({
+      functionName: 'getFeeCollector',
+      args: [batchPosters[1] as `0x${string}`],
```
Comment (chrstph-dvx) [inline src/decorators/arbAggregatorActions.integration.test.ts:56]: NIT: this should not be needed, type should be infered.

File: src/decorators/arbAggregatorActions.integration.test.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    const batchPosters = await nitroTestnodeL2Client.arbAggregatorReadContract({
+      functionName: 'getBatchPosters',
+    });
+
+    expect(batchPosters).toHaveLength(2);
+    expect(batchPosters[0]).toEqual('0xA4b000000000000000000073657175656e636572');
+
+    const batchPosterFeeCollector = await nitroTestnodeL2Client.arbAggregatorReadContract({
+      functionName: 'getFeeCollector',
+      args: [batchPosters[0] as `0x${string}`],
```
Comment (chrstph-dvx) [inline src/decorators/arbAggregatorActions.integration.test.ts:29]: This should not be needed now, return type should be infered (same below on L45 and 56)

**Review by chrstph-dvx — COMMENTED** [review]
Overall LGTM, just one minor comment.

**Review by chrstph-dvx — APPROVED** [review]

**Review by chrstph-dvx — APPROVED** [review]

**Pull request merged**